src/Elastic.Apm/ApmAgentExtensions.cs

```diff
@@ -23,7 +23,12 @@ public static class ApmAgentExtensions
 		public static IDisposable Subscribe(this IApmAgent agent, params IDiagnosticsSubscriber[] subscribers)
 		{
 			var subs = subscribers ?? Array.Empty<IDiagnosticsSubscriber>();
-			return subs.Aggregate(new CompositeDisposable(), (d, s) => d.Add(s.Subscribe(agent)));
+			var retVal = subs.Aggregate(new CompositeDisposable(), (d, s) => d.Add(s.Subscribe(agent)));
+
+			if (agent is ApmAgent apmAgent)
```

Comment:
nitpick `is IDisposable d)` since that is all we care about?

Reply:
👍

Reply:
Update: unfortunately that doesn't work, because the `ApmAgent` class collects these disposables, so we have to cast to `ApmAgent`. Due to this change the `ApmAgent` collects the return values form the `Subscribe` method, and with that we can `Dispose` the `ApmAgent` instance in tests to unsubscribe from all diagnostic source events.